### ISSUE TITLE
New release for CVE-2024-24790

### ISSUE THREAD

**Issue by giovannirco** [hypnoglow/helm-s3#434]
The current version is vulnerable to [CVE-2024-24790 ](https://avd.aquasec.com/nvd/cve-2024-24790)

```
root/.local/share/helm/plugins/helm-s3.git/bin/helm-s3 (gobinary)
=================================================================
Total: 1 (CRITICAL: 1)
┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.19.13           │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

Are there any plans for releasing a new version fixing it?

**Comment by giovannirco** [comment]
I have jsut tried using this version and still get the vulnerability, it reports go 1.22.0 installed, what am I missing?

```
root/.local/share/helm/plugins/helm-s3.git/bin/helm-s3 (gobinary)
=================================================================
Total: 1 (CRITICAL: 1)
┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.22.0            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

**Comment by hypnoglow** [comment]
I don't think that the fact the binary was build using 1.22.0 is enough to say that it is vulnerable.

**Comment by weinguy-env0** [comment]
I'm also still getting it in the scans. 
Can we bump to the latest patch version when building the binary so we won't get false positives?

**Comment by hypnoglow** [comment]
[v0.16.2](https://github.com/hypnoglow/helm-s3/releases/tag/v0.16.2) is now released with Go updated to 1.22.5
